src/Raven.Server/Documents/Handlers/Processors/Studio/StudioCollectionHandlerProcessorForDeleteCollection.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Operations;
+using Raven.Server.Documents.Operations;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Web.Studio;
+
+namespace Raven.Server.Documents.Handlers.Processors.Studio
+{
+    internal class StudioCollectionHandlerProcessorForDeleteCollection : AbstractStudioCollectionHandlerProcessorForDeleteCollection<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        public StudioCollectionHandlerProcessorForDeleteCollection([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override async ValueTask DeleteCollectionAsync(DocumentsOperationContext context, IDisposable returnContextToPool, string collectionName, HashSet<string> excludeIds, long operationId)
+        {
+            await ExecuteCollectionOperationAsync((runner, collectionNameParam, options, onProgress, token) => Task.Run(async () => await runner.ExecuteDelete(collectionNameParam, 0, long.MaxValue, options, onProgress, token)),
+                context, returnContextToPool, OperationType.DeleteByCollection, collectionName, operationId, excludeIds);
+        }
+
+        protected override long GetNextOperationId()
+        {
+            return RequestHandler.Database.Operations.GetNextOperationId();
+        }
+
+        private async Task ExecuteCollectionOperationAsync(Func<CollectionRunner, string, CollectionOperationOptions, Action<IOperationProgress>, OperationCancelToken, Task<IOperationResult>> operation, DocumentsOperationContext docsContext, IDisposable returnContextToPool, OperationType operationType, string collectionName, long operationId, HashSet<string> excludeIds)
```

Comment:
This looks like a sync method, you can remove 'async'

Reply:
removed and turned into void